library/core/src/Header.php

```diff
@@ -0,0 +1,223 @@
+        $links = [];
+
+        if ($script) {
+            $script = self::parsePlaceholders($script);
+            $path = self::createFullPath($basePath, $script);
+            $scripts[] = self::createElement($path, 'script');
+        }
+
+        if ($link) {
+            $link = self::parsePlaceholders($link);
+            $path = self::createFullPath($basePath, $link);
+            $links[] = self::createElement($path, 'link');
+        }
+
+        return ['scripts' => $scripts, 'links' => $links];
+    }
+
+    static private function loadTheme()
+    {
+        return self::createElement($GLOBALS['css_header'], 'link');
+    }
+
+    /**
+     * Parse a string for $GLOBAL key placeholders %key-name%.
+     *
+     * Perform a regex match all in the given subject for anything warpped in
+     * percent signs `%some-key%` and if that string exists in the $GLOBALS
+     * array, will replace the occurence with the value of that key.
+     *
+     * @param string $subject String containing placeholders (%key-name%)
+     * @return string The new string with properly replaced keys
+     */
+    static public function parsePlaceholders($subject)
+    {
+        $re = '/%(.*)%/';
+        $matches = [];
+        preg_match_all($re, $subject, $matches, PREG_SET_ORDER, 0);
+
+        foreach ($matches as $match) {
+            if (array_key_exists($match[1], $GLOBALS)) {
+                $subject = str_replace($match[0], $GLOBALS["{$match[1]}"], $subject);
+            }
+        }
+        return $subject;
+    }
+
+    /**
+     * Create the actual HTML element.
+     *
+     * @param string $path File path to load
+     * @param string $type Must be `script` or `link`
+     * @return string mixed HTML element
+     */
+    static private function createElement($path, $type)
+    {
+
+        $script = "<script type=\"text/javascript\" src=\"%path%\"></script>\n";
+        $link = "<link rel=\"stylesheet\" href=\"%path%\" type=\"text/css\">\n";
+
+        $template = ($type == 'script') ? $script : $link;
+        $v = $GLOBALS['v_js_includes'];
+        $path = $path . "?v={$v}";
+        return str_replace("%path%", $path, $template);
+
+    }
+
+    /**
+     * Create a full path from given parts.
+     *
+     * @param string $base Base path
+     * @param string $path specific path / filename
+     * @return string The full path
+     */
+    static private function createFullPath($base, $path)
+    {
+        return $base . $path;
+    }
+
+    /**
+     * Read a config file and turn it into an array.
+     *
+     * @param string $file Full path to filename
+     * @return array Array of assets
+     */
+    static private function readConfigFile($file)
+    {
+        try {
+            $config = Yaml::parse(file_get_contents($file));
+            return $config['assets'];
+        } catch (ParseException $e) {
+            error_log($e->getMessage());
+            // @TODO need to handle this better. RD 2017-05-24
+        }
+    }
```

Comment:
All this code makes my head spin a bit :)
The proof will be in the pudding (ie. testing will tell).